src/Psalm/Internal/Provider/ReturnTypeProvider/ArrayColumnReturnTypeProvider.php

```diff
@@ -114,6 +114,8 @@ public static function getFunctionReturnType(FunctionReturnTypeProviderEvent $ev
                 }
                 //array_column skips undefined elements so resulting type is necessarily defined
                 $result_element_type->possibly_undefined = false;
+            } elseif ($value_column_name === null) {
```

Comment:
One last thing, the fact that value_column_name is null here doesn't mean the second arg is null. It just means it wasn't a single Integer not a single String.
(For example, this test will fail `$d = array_column([["a" => 1], ["a" => 2], ["a" => 3]], rand(0,1)? 'a': 'b', "b");` with your code)

Could you create a new var value_column_name_is_null and set to true on line 86 when `$second_arg_type->isNull()` is true?

Reply:
Sure
I fixed this, and also add a test for that case